# Meta_Microbial-Workshop-2024
Official repository for the Meta_Microbial workshop which aims at providing an overview of current tools to analyze next-generation sequence data and to make insightful interpretation towards the fields of microbial ecology and environmental microbiology. 
This repository provides material and instruction accordingly to the lessons provided during the workshop and the sections belows are organized to mimic the day by day timeline.
Access to a remote machine will be available to ensure the correct 

# Access to the virtual machine - Instructions
- [ ] step 1
- [ ] step 2
- [ ] step n

# Local installation 
Necessary softwares:
+ software 1
+ software 2
+ software n

> [!NOTE]
> Useful information that users should know, even when skimming content.

> [!TIP]
> Helpful advice for doing things better or more easily.

> [!IMPORTANT]
> Key information users need to know to achieve their goal.

> [!WARNING]
> Urgent info that needs immediate user attention to avoid problems.

> [!CAUTION]
> Advises about risks or negative outcomes of certain actions.

eventual footnotes [^1]

[^1]: reference

# First Day
## Next-generation sequencing technologies and data generation
## Sampling protocols and standardization
## R tools for microbial ecology studies
## Python in microbial ecology studies
## Metabarcoding (amplicon sequencing) analysis workflow

# Second Day
## Exploring Online Resources and Repositories I
## Insights on data visualization and analysis
## Metagenomics (shotgun sequencing) analysis workflow
## Phylogenetic trees from high-throughput sequence data
## Exploring Online Resources and Repositories II
## Alternative statistical and data analyses

# Third Day
## Hands-on Metabarcoding</summary>
### From raw data to community structure insights
## Hands-on Metagenomics</summary>
### From raw data to taxonomic and functional insights 
